frontend/wallet/src/render/views/settings/CompressorSettings.re

```diff
@@ -0,0 +1,158 @@
+open Tc;
+
+module Styles = {
+  open Css;
+
+  let label =
+    merge([
+      Theme.Text.Body.semiBold,
+      style([color(Theme.Colors.midnight), marginBottom(`rem(0.25))]),
+    ]);
+};
+
+type modalState =
+  | Delegate(option(PublicKey.t), option(Int64.t))
+  | Stake(option(Int64.t));
+
+let intForState = state =>
+  switch (state) {
+  | Delegate(_, _) => 0
+  | Stake(_) => 1
+  };
+
+module CompressorModal = {
+  [@react.component]
+  let make = (~state, ~updateModal) =>
+    <Modal
+      title="Change consensus" onRequestClose={_ => updateModal(_ => None)}>
+      <Spacer height=0.5 />
+      <ToggleButton
+        options=[|"Delegate", "Stake"|]
+        selected={intForState(state)}
+        onChange={value =>
+          updateModal(_ =>
+            switch (value) {
+            | 0 => Some(Delegate(None, None))
+            | 1 => Some(Stake(None))
+            | _ => None
```

Comment:
will this ever happen?

Reply:
No, but it's future proofing and the type system doesn't let you write switch statements that aren't exhaustive, so we need to match all integers.

Reply:
gotcha, my thought was that we should just use a bool if there's only two values - but if you anticipate something else down the road, then this makes sense